packages/app/search/db.ts

```diff
@@ -16,87 +11,47 @@ class searchDb {
         }
     }
 
-    /* *  This function go to database and return a query with results.*/
-    async searchQuery(query: string): Promise<Model> {
-        const searchResults = await this.#db
-            .collection(MODELS_COLLECTION)
-            .find({ query })
-            .sort({ 'x-meditor.modifiedOn': -1 })
-            .limit(1)
-            .toArray()
-
-        return searchwithMonquery(searchResults)[0]
-    }
-
-    /* *  This function do search for specific model*/
-    async searchModel(modelName: string): Promise<Model> {
-        const models = await this.#db
-            .collection(MODELS_COLLECTION)
-            .find({ [MODELS_TITLE_PROPERTY]: modelName })
-            .sort({ 'x-meditor.modifiedOn': -1 })
-            .limit(1)
-            .toArray()
-
-        return makeSafeObjectIDs(models)[0]
-    }
-
-    async searchModels(): Promise<Model[]> {
-        const models = await this.#db
-            .collection(MODELS_COLLECTION)
-            .aggregate(
-                [
-                    { $sort: { 'x-meditor.modifiedOn': -1 } }, // Sort descending by version (date)
-                    { $group: { _id: '$name', doc: { $first: '$$ROOT' } } }, // Grab all fields in the most recent version
-                    { $replaceRoot: { newRoot: '$doc' } }, // Put all fields of the most recent doc back into root of the document
-                ],
-                { allowDiskUse: true }
-            )
-            .toArray()
-
-        return makeSafeObjectIDs(models)
+    compileQuery(query: string) {
+        return compile(query)
     }
 
-    async searchDocumentsForModel(
+    async search(
         modelName: string,
-        searchOptions: DocumentsSearchOptions,
-        titleProperty: string
-    ): Promise<Document[]> {
+        titleProperty: string,
+        query: string,
+        resultsPerPage: number,
+        pageNumber: number
+    ): Promise<any> {
+        //* The pipeline order matters: we have to sort and group by titleProperty first so that we're not matching old documents.
         const pipeline = [
-            // filter out deleted documents
-            {
-                $match: {
-                    'x-meditor.deletedOn': { $exists: false },
-                },
-            },
+            // Sort descending by version (date).
+            { $sort: { 'x-meditor.modifiedOn': -1 } },
+            // Grab all fields in the most recent version.
+            { $group: { _id: `$${titleProperty}`, doc: { $first: '$$ROOT' } } },
+            // Put all fields of the most recent doc back into root of the document.
+            { $replaceRoot: { newRoot: '$doc' } },
+            // compile Lucene syntax into MQL
+            { $match: this.compileQuery(query) },
```

Comment:
We still need the deletedOn exclusion here, so users aren't seeing "deleted" documents in their search results